lib/bolt/transport/local.rb

```diff
@@ -108,16 +121,43 @@ def run_task(target, task, arguments, _options = {})
           copy_file(executable, script)
           File.chmod(0o750, script)
 
-          # unpack any Sensitive data, write it to a separate variable because
-          # we log 'arguments' below
+          # log the arguments with sensitive data redacted, do NOT log unwrapped_arguments
+          logger.debug("Running '#{script}' with #{arguments}")
           unwrapped_arguments = unwrap_sensitive_args(arguments)
+
           stdin = STDIN_METHODS.include?(input_method) ? JSON.dump(unwrapped_arguments) : nil
-          env = ENVIRONMENT_METHODS.include?(input_method) ? envify_params(unwrapped_arguments) : nil
 
-          # log the arguments with sensitive data redacted, do NOT log unwrapped_arguments
-          logger.debug("Running '#{script}' with #{arguments}")
+          if Bolt::Util.windows?
+            # WINDOWS
+            if ENVIRONMENT_METHODS.include?(input_method)
+              environment_params = envify_params(unwrapped_arguments).each_with_object([]) do |(arg, val), list|
+                list << Powershell.set_env(arg, val)
+              end
+              environment_params = environment_params.join("\n") + "\n"
+            else
+              environment_params = ""
+            end
 
-          output = @conn.execute(script, stdin: stdin, env: env, dir: dir)
+            output =
+              if Powershell.powershell_file?(script) && stdin.nil?
+                command = Powershell.run_ps_task(arguments, script, input_method)
+                command = environment_params + Powershell.shell_init + command
+                if input_method == 'powershell'
+                  @conn.execute(command, dir: dir, env: "powershell.exe")
+                else
+                  @conn.execute(command, dir: dir, stdin: stdin, env: "powershell.exe")
```

Comment:
This feels like a weird use of `env`. `env` was supposed to be about setting up environment variables. However it looks like the way we order that in shell.rb doesn't work for Windows.